### PR TITLE
feat: add elb security policy

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
add new resource of huaweicloud_elb_security_policy
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=
'./huaweicloud/services/acceptance/elb' TESTARGS='-run=TestAccSecurityPoliciesV3_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb -v -run=TestAccSecurityPoliciesV3_basic -timeout 360m -parallel 4     
=== RUN   TestAccSecurityPoliciesV3_basic
=== PAUSE TestAccSecurityPoliciesV3_basic
=== CONT  TestAccSecurityPoliciesV3_basic
--- PASS: TestAccSecurityPoliciesV3_basic (17.58s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       17.636s
```

### CODE REVIEW

**Comment by chengxiangdong** [comment]
pls fill in the test results

**Comment by chengxiangdong** [comment]
pls fix the doc-lint and CI issues

**Comment by houpeng80** [comment]
> pls fill in the test results
make testacc TEST=
'./huaweicloud/services/acceptance/elb' TESTARGS='-run=TestAccSecurityPoliciesV3_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb -v -run=TestAccSecurityPoliciesV3_basic -timeout 360m -parallel 4     
=== RUN   TestAccSecurityPoliciesV3_basic
=== PAUSE TestAccSecurityPoliciesV3_basic
=== CONT  TestAccSecurityPoliciesV3_basic
--- PASS: TestAccSecurityPoliciesV3_basic (17.58s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       17.636s

File: docs/resources/elb_security_policy.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ciphers               = ["ECDHE-RSA-AES256-GCM-SHA384","ECDHE-RSA-AES128-GCM-SHA256"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
```
Comment (deer-hang) [inline docs/resources/elb_security_policy.md:26]: Changing this creates a new security_policy.

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+func SecurityPoliciesV3ListenerRefSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: return &schema.Resource{
		Schema: map[string]*schema.Schema{
			"id": {
				Type:     schema.TypeString,
				Computed: true,
			},
		},
	}

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	d.SetId(id.(string))
+
+	return resourceSecurityPoliciesV3Read(ctx, d, meta)
+}
+
+func buildCreateSecurityPolicyBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"security_policy": buildCreateSecurityPolicyChildBodyParams(d, config),
+	}
+	return bodyParams
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: return map[string]interface{}{
		"security_policy": buildCreateSecurityPolicyChildBodyParams(d, config),
	}

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+func buildCreateSecurityPolicyChildBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":                  utils.ValueIngoreEmpty(d.Get("name")),
+		"description":           utils.ValueIngoreEmpty(d.Get("description")),
+		"enterprise_project_id": utils.ValueIngoreEmpty(common.GetEnterpriseProjectID(d, config)),
+		"protocols":             utils.ValueIngoreEmpty(d.Get("protocols")),
+		"ciphers":               utils.ValueIngoreEmpty(d.Get("ciphers")),
+	}
+	return bodyParams
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+		}
+	}
+	return resourceSecurityPoliciesV3Read(ctx, d, meta)
+}
+
+func buildUpdateSecurityPolicyBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"security_policy": buildUpdateSecurityPolicyChildBodyParams(d, config),
+	}
+	return bodyParams
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+}
+
+func buildUpdateSecurityPolicyChildBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"protocols":   utils.ValueIngoreEmpty(d.Get("protocols")),
+		"ciphers":     utils.ValueIngoreEmpty(d.Get("ciphers")),
+	}
+	return bodyParams
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 6 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: docs/resources/elb_security_policy.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
+---
+subcategory: "Dedicated Load Balance (Dedicated ELB)"
+---
+
+# huaweicloud_elb_security_policy
+
+Manages an ELB security policies resource within HuaweiCloud.
```
Comment (hmbig2) [inline docs/resources/elb_security_policy.md]:
```suggestion
Manages an ELB security policy resource within HuaweiCloud.
```

File: docs/resources/elb_security_policy.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `protocols` - (Required, List) Specifies the TSL protocol list which the security policy select.
+
+* `ciphers` - (Required, List) Specifies the cipher suite list of the security policy.
+  The protocol and cipher suite must match. That is to say, there must be at least one cipher suite in
+  ciphers that matches the protocol.
+
+* `name` - (Optional, String) Specifies the ELB security policy name.
+  The name contains only letters, digits, underscores (_), and hyphens (-), and cannot exceed 64 characters.
```
Comment (hmbig2) [inline docs/resources/elb_security_policy.md]: add regx pattern to schema

File: docs/resources/elb_security_policy.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+* `description` - (Optional, String) Specifies the description of the ELB security policy.
+  The value can contain 0 to 255 characters.
```
Comment (hmbig2) [inline docs/resources/elb_security_policy.md:47]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"protocols": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Required: true,
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: Did you delete the description?

File: docs/resources/elb_security_policy.md (hunk 7 above)
Comment (houpeng80) [reply docs/resources/elb_security_policy.md]: done

File: docs/resources/elb_security_policy.md (hunk 8 above)
Comment (houpeng80) [reply docs/resources/elb_security_policy.md]: done

File: docs/resources/elb_security_policy.md (hunk 9 above)
Comment (houpeng80) [reply docs/resources/elb_security_policy.md:47]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 10 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+
+	"github.com/chnsz/golangsdk"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const (
+	stringRegxp     = `^[\x{4E00}-\x{9FFC}A-Za-z-_0-9?'#().,&%@!]*$`
+	stringFormatMsg = "Only letters, Chinese characters, digits, hyphens (-), underscores (_) and" +
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: this regxp is different with doc

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The enterprise project ID to which the Enterprise router belongs.`,
+			},
+			"listeners": {
+				Type:        schema.TypeList,
+				Elem:        SecurityPoliciesV3ListenerRefSchema(),
+				Computed:    true,
+				Description: `the listener which the security policy associated with.`,
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]:
```suggestion
				Description: `The listener which the security policy associated with.`,
```

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+					validation.StringMatch(regexp.MustCompile(stringRegxp), stringFormatMsg),
+				),
+				Description: `Specifies the description of the ELB security policy`,
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The enterprise project ID to which the Enterprise router belongs.`,
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: keep the uniform format, add `Specifies` to this attr, or remove `Specifies` of others.
Same to doc

Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 12 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go (hunk 11 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: done

File: docs/resources/elb_security_policy.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+Manages an ELB security policy resource within HuaweiCloud.
+
+## Example Usage
+
+```HCL
+resource "huaweicloud_elb_security_policy" "test" {
+name                  = "security_policy_test"
+description           = "this is a security policy"
+enterprise_project_id = "4579f2f5-cbe8-425a-8f32-53dcb9d9053a"
```
Comment (ShiChangkuo) [inline docs/resources/elb_security_policy.md]: suggest to put the `enterprise_project_id` at the end of this resource or delete it.

File: docs/resources/elb_security_policy.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+  ciphers that matches the protocol.
+
+* `name` - (Optional, String) Specifies the ELB security policy name.
+  The name contains only Chinese characters, letters, digits, underscores (_), and hyphens (-),
+  and cannot exceed 255 characters.
+
+* `description` - (Optional, String) Specifies the description of the ELB security policy.
+  The value can contain 0 to 255 characters.
+
+* `enterprise_project_id` - (Optional, String, ForceNew) Specifies The enterprise project ID to which the Enterprise
```
Comment (ShiChangkuo) [inline docs/resources/elb_security_policy.md]:
```suggestion
* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID to which the Enterprise
```

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+		},
+	})
+}
+
+func testSecurityPoliciesV3_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_elb_security_policy" "test" {
+  protocols = ["TLSv1.1","TLSv1.2"]
+  ciphers = ["ECDHE-ECDSA-AES128-SHA", "ECDHE-RSA-AES256-SHA"]
+  name = "%s"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go]: format error, please align with the equal sign

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+}
+`, name)
+}
+
+func testSecurityPoliciesV3_basic_update(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_elb_security_policy" "test" {
+  protocols = ["TLSv1.2"]
+  ciphers = ["ECDHE-ECDSA-AES128-SHA"]
+  name = "%s"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go]: ditto

File: docs/resources/elb_security_policy.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `protocols` - (Required, List) Specifies the TSL protocol list which the security policy select.
```
Comment (ShiChangkuo) [inline docs/resources/elb_security_policy.md:27]: please add valid values or add a validation in schema.

File: docs/resources/elb_security_policy.md (hunk 14 above)
Comment (houpeng80) [reply docs/resources/elb_security_policy.md]: done

File: docs/resources/elb_security_policy.md (hunk 15 above)
Comment (houpeng80) [reply docs/resources/elb_security_policy.md]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go (hunk 16 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go (hunk 17 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_security_policy_test.go]: done

File: docs/resources/elb_security_policy.md (hunk 18 above)
Comment (houpeng80) [reply docs/resources/elb_security_policy.md:27]: done

**Pull request merged**